File: src/main/java/fr/cph/crypto/domain/Ticker.java
package fr.cph.crypto.domain;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
public class Ticker implements Serializable {

	@Id
	private String id;
	private Currency currency;
	private Double price;
	private Double percentChange1h;
	private Double percentChange24h;
	private Double percentChange7d;
	private String lastUpdated; // change to date
}

File: src/main/kotlin/fr/cph/crypto/domain/Ticker.kt
package fr.cph.crypto.domain

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("id", "currency", "price", "percentChange1h", "percentChange24h", "percentChange7d", "lastUpdated")
@Document
data class Ticker(val currency: Currency,
                  val price: Double,
                  val percentChange1h: Double,
                  val percentChange24h: Double,
                  val percentChange7d: Double,
                  val lastUpdated: String) : Serializable {

    @Id
    var id: String? = null
}
